### ISSUE TITLE
Validate returns "No such file or directory: 'inline'" for all Stream errors

### ISSUE THREAD

**Issue by mattmelton** [frictionlessdata/goodtables-py#349]
# Overview

I need to validate CSV data from an existing stream. This was frustrating experience because I kept receiving the same error for several underlying issues.

After stepping through the code I got the following unit test to work:

```
    with open("unit-test-0123.csv", "rb") as source:
        report = validate(source, format='csv', schema=schema)
```

I discovered that:

1. streams must be binary
2. you must specify the format (streams are opaque)
3. you cannot load files in tmp because it is marked unsafe

Rather than see these errors the only exception raised was: `"No such file or directory: 'inline'"`. This occurs due to `inspector.py`'s catch-all handling of sources without a schema. In particular the catch block looks like this:

```        
...
   stream.open()
...
except Exception as exception:
    if _local_file_not_found(source):
        message = "No such file or directory: '%s'" % source
        exception = tabulator.exceptions.IOError(message)
```

and `_local_file_not_found` is defined as:

```
def _local_file_not_found(source):
    return urlparse(source).scheme == '' \
           and not os.path.isfile(source)
```

In memory streams lack a source URI - it is hardcoded to `inline`. 

And finally, warnings such as the one issued when the max error count is reached confusingly refer to a table called `inline`:

`Table "inline" inspection has reached 5 error(s) limit`

---

Please preserve this line to notify @roll (lead of this repository)

**Comment by roll** [comment]
Thanks. We need to improve the way it reports this error

**Comment by roll** [comment]
FIXED in #364
